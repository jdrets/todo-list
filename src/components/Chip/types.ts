export interface ChipTypes {
  onClose?: () => void
  label: string
}
