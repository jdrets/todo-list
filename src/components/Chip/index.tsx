import React, { FunctionComponent } from 'react'
import { Close } from 'react-ionicons'

import { ChipTypes } from './types'
import {
  Wrapper,
  Container,
  Label,
  CloseButton
} from './styles'

const Chip: FunctionComponent<ChipTypes> = ({
  deleteFilter,
  filter
}) => {
  const handleClose = () => {
    deleteFilter(filter[0])
  }

  return (
    filter[1] && <Wrapper>
      <Container>
        <Label>{filter[0]}: {filter[1]}</Label>
        <CloseButton onClick={handleClose} data-testid="chip-close-button">
          <Close/>
        </CloseButton>
      </Container>
    </Wrapper>
  )
}

export default Chip
