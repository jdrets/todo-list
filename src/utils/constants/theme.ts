const palette = {
  purpleDark: '#4B00BC',
  purple: '#6600FF',
  purpleMedium: '#8533FF',
  purpleLight: '#E3D0FF',
  green: '#59BC6F',
  greenLight: '#E6F5E2',
  red: '#EB5757',
  redLight: '#FCE4E4',
  yellow: '#DD9900',
  yellowLight: '#FEF1D8',
  white: '#FFFFFF',
  black: '#000000',
  gray: '#898989',
  grayLight: '#E0E0E0'
}

const theme = {
  colors: {
    primaryColor: palette.purple,
    primaryColorMedium: palette.purpleMedium,
    primaryColorLight: palette.purpleLight,
    primaryColorDark: palette.purpleDark,
    ...palette
  },
  layout: {
    breakpoints: {
      medium: '768px'
    },
    mobile: {
      width: '100%',
      padding: '0 24px'
    },
    desktop: {
      width: '1050px'
    }
  }
}

export default theme
