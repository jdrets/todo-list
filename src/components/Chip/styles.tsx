import styled from 'styled-components'

import theme from '../../utils/constants/theme'

export const Wrapper = styled.div`
  background-color: ${theme.colors.primaryColor};
  border-radius: 24px;
  color: white;
  cursor: pointer;
  display: inline-block;
  margin: 0 4px 4px 0;
  padding: 6px;
  transition: 0.3s;
`

export const Container = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
`

export const Label = styled.div`
  font-size: 10px;
  font-weight: 500;
  padding: 0 8px;
`

export const CloseButton = styled.div`
  background-color: ${theme.colors.primaryColorDark};
  border-radius: 24px;
  font-size: 10px;
  
  svg {
    align-items: center;
    display: flex;
    fill: white;  
    height: 14px;
    width: 14px;
  }
  
  &:hover {
    background-color: ${theme.colors.primaryColor};
  }
`
