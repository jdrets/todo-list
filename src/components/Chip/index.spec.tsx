import React from 'react'
import { ThemeProvider } from 'styled-components'

import { render, fireEvent } from '@testing-library/react'

import Chip from '.'
import theme from '../../utils/constants/theme'

const setup = (mockData = {}) => {
  const mock = {
    onClose: jest.fn(),
    label: 'dummy label'
  }
  return render(
    <ThemeProvider theme={theme}>
      <Chip {...mock} {...mockData}>
        Button
      </Chip>
    </ThemeProvider>
  )
}

describe('<Chip />', () => {
  test('should match with snapshot', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })

  test('should call onClose function when close button is clicked', () => {
    const mockOnClose = jest.fn()
    const mockData = {
      onClose: mockOnClose
    }
    const { getByTestId } = setup(mockData)
    const button = getByTestId('chip-close-button')

    fireEvent.click(button)

    expect(mockOnClose).toHaveBeenCalled()
  })
})
