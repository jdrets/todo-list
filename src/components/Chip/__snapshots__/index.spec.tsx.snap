// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> should match with snapshot 1`] = `
<div>
  <div
    class="sc-gtssRu dngUbG"
  >
    <div
      class="sc-dlnjPT bmVKDB"
    >
      <div
        class="sc-hKFyIo jmlARW"
      >
        dummy label
      </div>
      <div
        class="sc-eCApGN FQpIN"
        data-testid="chip-close-button"
      >
        <span
          class="sc-bdnylx kpuqbH"
          color="#000000"
          height="22px"
          rotate="0"
          width="22px"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
